packages/xo-server/src/xapi/mixins/vm.js

```diff
@@ -290,6 +290,17 @@ export default {
       },
     },
 
+    blockedOperations: {
+      set(operations, vm) {
+        return this.setFieldEntries(
```

Comment:
You can use `vm.update_blocked_operations`

Reply:
Why? `setFieldEntries` handles `null`, duplicate keys, etc. which is what I need.

Reply:
`vm.update_blocked_operations` will call `setFieldEntries`. Please see https://github.com/vatesfr/xen-orchestra/blob/master/packages/xen-api/src/index.js#L1162